integrations/gmail.py

```
@@ -0,0 +1,239 @@
+                # filter for invoice emails
+                invoice_emails = [email for email in emails if email["has_invoice"]]
+                
+                if invoice_emails:
+                    # process the emails with the callback
+                    await callback(invoice_emails)
+                
+                # disconnect
+                await self.disconnect()
+                
+            except Exception as e:
+                logger.error(f"Error monitoring inbox: {e}", exc_info=True)
```

Ruff (0.12.2)

Undefined name `logger`

(F821)